src/gui/base/Dialog.ts

```diff
@@ -772,6 +765,47 @@ export class Dialog implements ModalComponent {
 		})
 	}
 
+	/**
+	 * Shows a dialog with a text field input and ok/cancel buttons. The text input is processed in the okayAction.
+	 * @param titleId title of the dialog
+	 * @param labelIdOrLabelFunction label of the text field
+	 * @param infoMsgId help label of the text field
+	 * @param value initial value
+	 * @param inputValidator Called when "Ok" is clicked receiving the entered text. Must return null if the text is valid or an error messageId if the text is invalid, so an error message is shown.
+	 * @param okAction Called when "OK" is clicked, receives the entered text. If the okayAction results in a ConnectionsError the dialog is not closed.
+	 */
+	static showProcessTextInputDialog(
```

Comment:
The comment should say at the top how this is different from the textInputDialog (currently only hinted at in the okAction param description)